spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/AddressProviderConfiguration.java

```diff
@@ -0,0 +1,45 @@
+package org.springframework.cloud.zookeeper.discovery;
+
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+import static java.lang.invoke.MethodHandles.lookup;
+
+/**
+ * Configuration that registers a bean related to microservice's address and port providing.
+ *
+ * @see MicroserviceAddressProvider
+ */
+@Configuration
+public class AddressProviderConfiguration {
+
+    private static final Logger log = LoggerFactory.getLogger(lookup().lookupClass());
+
+    @Autowired
+    private Environment environment;
+
+    @Bean
+    MicroserviceAddressProvider microserviceAddressProvider() {
+        String microserviceHost = environment.getProperty("microservice.host", resolveMicroserviceLocalhost());
+        Integer microservicePort = Integer.valueOf(environment.getProperty("server.port", "8080"));
+        return new MicroserviceAddressProvider(microserviceHost, microservicePort);
+    }
+
+    public static String resolveMicroserviceLocalhost() {
+        try {
+            return InetAddress.getLocalHost().getHostAddress();
```

Comment:
I'm not too familiar with the intended usage of `MircoserviceAddressProvider`, so feel free to correct me if I'm wrong. It seems the intent is to publish the local address and port of a service to ZK for service discovery.

On some operating systems (Linux in my experience) `InetAddress.getLocalHost().getHostAddress()` will return the loopback address 127.0.0.1, which won't be useful for external clients. This is discussed here:

http://stackoverflow.com/questions/2381316/java-inetaddress-getlocalhost-returns-127-0-0-1-how-to-get-real-ip

In Spring XD this is how we "solve" the problem:

https://github.com/spring-projects/spring-xd/blob/6bd17162c8a6da0f09f6f8809f694a060c71ecc0/spring-xd-dirt/src/main/java/org/springframework/xd/dirt/util/RuntimeUtils.java#L66

Most of the time this will work if there's a single NIC, but if there are multiple NICs then it may not return the one you'd like.